server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1104,7 +1104,8 @@ static boolean supportsParallelCollection(ResultsType resultsType, SearchSourceB
         /*
         TODO uncomment this block to enable inter-segment concurrency for the query phase
         if (resultsType == ResultsType.QUERY) {
-            return source == null || source.aggregations() == null || source.aggregations().supportsParallelCollection();
+            return source == null || source.collapse() == null || source.aggregations() == null
+            || source.aggregations().supportsParallelCollection();
```

Comment:
Perhaps yes, but it's a single situation for top docs collection compared to aggs where some of them don't support concurrency. I felt like it may be overkill to add a supportParallelCollection to SearchSourceBuilder. Maybe though that's a good idea, let me play with it.

Reply:
well, this made me realize that the conditional was wrong :) and that if I move it to SearchSourceBuilder I can test it separately from enabling parallel collection in the query phase (this method in SearchService can stay unchanged).

Reply:
Here you go: https://github.com/elastic/elasticsearch/pull/98425/commits/d239f0bac1d58fccd5828e0e6b54d79fffef7927